Libs/MRML/DisplayableManager/vtkMRMLSliceIntersectionInteractionRepresentation.cxx

```diff
@@ -0,0 +1,2333 @@
+  vtkNew<vtkMatrix4x4> sliceOffsetHandle1ToWorldMatrix, sliceOffsetHandle2ToWorldMatrix; // Compute transformation matrix (rotation + translation)
+  this->ComputeHandleToWorldTransformMatrix(sliceOffsetHandle1Position, sliceOffsetHandleOrientation2D, sliceOffsetHandle1ToWorldMatrix);
+  this->ComputeHandleToWorldTransformMatrix(sliceOffsetHandle2Position, sliceOffsetHandleOrientation2D, sliceOffsetHandle2ToWorldMatrix);
+  pipeline->SliceOffsetHandle1ToWorldTransform->Identity();
+  pipeline->SliceOffsetHandle1ToWorldTransform->SetMatrix(sliceOffsetHandle1ToWorldMatrix); // Update handles to world transform
+  pipeline->SliceOffsetHandle2ToWorldTransform->Identity();
+  pipeline->SliceOffsetHandle2ToWorldTransform->SetMatrix(sliceOffsetHandle2ToWorldMatrix); // Update handles to world transform
+
+  // Update slice offset handle interaction points
+  vtkNew<vtkPoints> sliceOffsetHandlePoints;
+  double sliceOffsetHandlePoint[3] = { 0.0, 0.0, 0.0 };
+  double sliceOffsetHandlePoint_h[4] = { 0.0, 0.0, 0.0, 1.0 }; //homogeneous coordinates
+  if (pipeline->SliceOffsetHandle1Visible) // Handle 1
+    {
+    for (int i = 0; i < pipeline->SliceOffsetHandle1Points->GetNumberOfPoints(); i++)
+      {
+      pipeline->SliceOffsetHandle1Points->GetPoint(i, sliceOffsetHandlePoint);
+      sliceOffsetHandlePoint_h[0] = sliceOffsetHandlePoint[0];
+      sliceOffsetHandlePoint_h[1] = sliceOffsetHandlePoint[1];
+      sliceOffsetHandlePoint_h[2] = sliceOffsetHandlePoint[2];
+      sliceOffsetHandle1ToWorldMatrix->MultiplyPoint(sliceOffsetHandlePoint_h, sliceOffsetHandlePoint_h);
+      sliceOffsetHandlePoint[0] = sliceOffsetHandlePoint_h[0];
+      sliceOffsetHandlePoint[1] = sliceOffsetHandlePoint_h[1];
+      sliceOffsetHandlePoint[2] = sliceOffsetHandlePoint_h[2];
+      sliceOffsetHandlePoints->InsertNextPoint(sliceOffsetHandlePoint);
+      }
+    }
+  if (pipeline->SliceOffsetHandle2Visible) // Handle 2
+    {
+    for (int j = 0; j < pipeline->SliceOffsetHandle2Points->GetNumberOfPoints(); j++)
+      {
+      pipeline->SliceOffsetHandle2Points->GetPoint(j, sliceOffsetHandlePoint);
+      sliceOffsetHandlePoint_h[0] = sliceOffsetHandlePoint[0];
+      sliceOffsetHandlePoint_h[1] = sliceOffsetHandlePoint[1];
+      sliceOffsetHandlePoint_h[2] = sliceOffsetHandlePoint[2];
+      sliceOffsetHandle2ToWorldMatrix->MultiplyPoint(sliceOffsetHandlePoint_h, sliceOffsetHandlePoint_h);
+      sliceOffsetHandlePoint[0] = sliceOffsetHandlePoint_h[0];
+      sliceOffsetHandlePoint[1] = sliceOffsetHandlePoint_h[1];
+      sliceOffsetHandlePoint[2] = sliceOffsetHandlePoint_h[2];
+      sliceOffsetHandlePoints->InsertNextPoint(sliceOffsetHandlePoint);
+      }
+    }
+
+  // Define points along intersection lines for interaction
+  vtkPoints* line1PointsDefault;
+  vtkPoints* line2PointsDefault;
+  line1PointsDefault = pipeline->IntersectionLine1->GetOutput()->GetPoints();
```

Comment:
Thanks for pointing out this issue. It seems that when the slice controller widget is used, the display pipelines which render the intersection lines do not correctly update. This seems to be the cause of this bug.

I have done some testing and I was able to solve this issue. I modified the function **trackSliceOffsetValue** of the file **qMRMLSliceControllerWidget.cxx**. This function is called whenever the slider value changes. I added some code to this function to call the "_Modified()_" function of every slice node in the scene, so that this happens when the slider value changes. Whenever the slice node is modified, the display pipelines are updated by recomputing the slice intersection point and line positions.

Doing this, the visualization of the intersection lines works well for those cases when slices are oblique and slice offset is modified using the sliders. I did not commit these changes, since I am not sure if this approach is the most convenient. However, this is useful to demonstrate that this issue could be solved.

Reply:
I have recorded a video of the interactive slice intersection using oblique slices. As you can see, the display pipelines are now updated when the slider is modified:

https://user-images.githubusercontent.com/10816661/141790195-f9690303-0dee-4013-92db-6106deb56e2f.mp4

What do you think?

Reply:
I removed the extra modified events to avoid slowdowns in slice browsing. Now, since the slice logics are observed, the display pipelines can be updated whenever the slider is used to change the current slice.